frontend/src/metabase/components/Header.jsx

```diff
@@ -39,104 +82,133 @@ export default class Header extends Component {
+                    onCancel={this.props.onHeaderModalCancel}
+                />
+                <div className="py1 lg-py2 xl-py3 wrapper flex align-center" ref="header">
                     <div className="Entity py3">
-                        {titleAndDescription}
-                        {attribution}
+                        {
+                            isEditingInfo
+                                ? (
+                                    <div className="Header-title flex flex-column flex-full bordered rounded my1">
+                                        <Input
+                                            className="AdminInput text-bold border-bottom rounded-top h3"
+                                            type="text"
+                                            value={item.name || ""}
+                                            onChange={this.setItemAttribute.bind(this, "name")}
+                                        />
+                                        <Input
+                                            className="AdminInput rounded-bottom h4"
+                                            type="text"
+                                            value={item.description || ""}
+                                            onChange={this.setItemAttribute.bind(this, "description")}
+                                            placeholder="No description yet"
+                                        />
+                                    </div>
+                                )
+                                : (
+                                    <TitleAndDescription
+                                        title={item && item.id
+                                                ? item.name
+                                                : `New ${objectType}`
+                                        }
+                                        description={item.description}
+                                    />
+                                )
+                        }
+                        {
+                            // TODO - @kdoh - does this ever evalutate to true?
+                            item && item.creator && (
+                                <div className="Header-attribution">
+                                    Asked by {item.creator.common_name}
+                                </div>
+                            )
+                        }
                     </div>
 
-                    <div className="flex align-center flex-align-right">
-                        {headerButtons}
+                    { /* TODO - this should get cleaned up */ }
+                    <div className="flex align-center ml-auto">
+                        {!isEditing && !isFullscreen && defaultActions.map((section, sectionIndex) => {
+                            return section && section.length > 0 && (
+                                <span key={sectionIndex} className="flex align-center">
```

Comment:
it's unclear to me why we need multiple sections

Reply:
Yeah I think that's vestigial. This part of the component still needs some love.